tests/test_input.py

```diff
@@ -1008,12 +1012,167 @@ def test_config_parsing_relative_input_paths():
      fixed_parameter_list, param_grid_list, featureset_names, learners,
      prediction_dir, log_path, train_path, test_path, ids_to_floats,
      class_map, custom_learner_path, learning_curve_cv_folds_list, learning_curve_train_sizes) = _parse_config_file(config_path)
 
 
+def test_cv_folds_and_grid_search_folds():
+
+    # we want to test all possible combinations of the following variables:
+    #  task = train, cross_validate
+    #  cv_folds/folds_file = not specified, number, csv file
+    #  grid_search_folds = not specified, number
+    #  use_folds_file_for_grid_search = not specified, True, False
+
+    # below is a table of what we expect for each of the combinations
+    # note: `fold_mapping` refers to the dictionary version of the folds file
+
+    # task, cv_folds/folds_file, grid_search_folds, use_folds_file_for_grid_search -> cv_folds, grid_search_folds
+    # ('train', None, None, None) ->  (None, 3)
+    # ('train', None, None, True) ->  (None, 3)
+    # ('train', None, None, False) ->  (None, 3)
+    # ('train', None, 7, None) ->  (None, 7)
+    # ('train', None, 7, True) ->  (None, 7)
+    # ('train', None, 7, False) ->  (None, 7)
+    # ('train', 5, None, None) ->  (None, 3)
+    # ('train', 5, None, True) ->  (None. 3)
+    # ('train', 5, None, False) ->   (None, 3)
+    # ('train', 5, 7, None) ->  (None, 7)
+    # ('train', 5, 7, True) ->  (None, 7)
+    # ('train', 5, 7, False) ->  (None, 7)
+    # ('train', 'train/folds_file_test.csv', None, None) ->  (None, fold_mapping)
+    # ('train', 'train/folds_file_test.csv', None, True) ->  (None, fold_mapping)
+    # ('train', 'train/folds_file_test.csv', None, False) ->  (None, fold_mapping)
+    # ('train', 'train/folds_file_test.csv', 7, None) ->  (None, fold_mapping)
+    # ('train', 'train/folds_file_test.csv', 7, True) ->  (None, fold_mapping)
+    # ('train', 'train/folds_file_test.csv', 7, False) ->  (None, fold_mapping)
+    # ('cross_validate', None, None, None) ->  (10, 3)
+    # ('cross_validate', None, None, True) ->  (10, 3)
+    # ('cross_validate', None, None, False) ->  (10, 3)
+    # ('cross_validate', None, 7, None) ->  (10, 7)
+    # ('cross_validate', None, 7, True) ->  (10, 7)
+    # ('cross_validate', None, 7, False) ->  (10, 7)
+    # ('cross_validate', 5, None, None) ->  (5, 3)
+    # ('cross_validate', 5, None, True) ->  (5, 3)
+    # ('cross_validate', 5, None, False) ->  (5, 3)
+    # ('cross_validate', 5, 7, None) ->  (5, 7)
+    # ('cross_validate', 5, 7, True) ->  (5, 7)
+    # ('cross_validate', 5, 7, False) ->  (5, 7)
+    # ('cross_validate', 'train/folds_file_test.csv', None, None) ->  (fold_mapping, fold_mapping)
+    # ('cross_validate', 'train/folds_file_test.csv', None, True) ->  (fold_mapping, fold_mapping)
+    # ('cross_validate', 'train/folds_file_test.csv', None, False) ->  (fold_mapping, 3)
+    # ('cross_validate', 'train/folds_file_test.csv', 7, None) ->  (fold_mapping, fold_mapping)
```

Comment:
Warning?

Reply:
@aloukina if you look in the code, I added some new warnings in `config.py`. This is a test, so not sure why we need warnings here?

Reply:
@aoifecahill here are [the](https://github.com/EducationalTestingService/skll/blob/fix/cv-folds-file-is-slower/skll/config.py#L632) [new](https://github.com/EducationalTestingService/skll/blob/fix/cv-folds-file-is-slower/skll/config.py#L635) [warnings](https://github.com/EducationalTestingService/skll/blob/fix/cv-folds-file-is-slower/skll/config.py#L644) that are already in the PR (each word is a link).

Reply:
You don't need them in test of course - it's just easier to see all places with warnings when you have all combinations written out with expected behaviour.
 Since you know the code better, you know where the warnings are already in place. For others we could just add an issue to add them in the future.

Reply:
I checked and I think I already have the warnings where I think they are really important.